dev/benchmark/sysbench-in-k8s.md

```diff
@@ -0,0 +1,460 @@
+---
+title: TiDB in Kubernetes Sysbench Performance Test
+summary: Learn the Sysbench performance test of TiDB in Kubernetes.
+category: benchmark
+---
+
+# TiDB in Kubernetes Sysbench Performance Test
+
+Since the release of [TiDB Operator GA](https://pingcap.com/blog/database-cluster-deployment-and-management-made-easy-with-kubernetes/), more users begin to deploy and manage the TiDB cluster in Kubernetes using TiDB Operator. In this report, an in-depth and comprehensive test of TiDB has been conducted on GKE, which offers insight into the influencing factors that affects the performance of TiDB in Kubernetes.
+
+## Test purpose
+
+- To test the performance of TiDB on a typical public cloud platform
+- To test the influences that the public cloud platform, network, CPU and different Pod networks have on the performance of TiDB
+
+## Test environment
+
+### Version and configuration
+
+In this test:
+
+- TiDB 3.0.1 and TiDB Operator 1.0.0 are used.
+- Three instances are deployed for PD, TiDB and TiKV respectively.
+- Each component is configured as below. Unconfigured components use the default values.
+
+PD:
+
+```toml
+[log]
+level = "info"
+[replication]
+location-labels = ["region", "zone", "rack", "host"]
+```
+
+TiDB:
+
+```toml
+[log]
+level = "error"
+[prepared-plan-cache]
+enabled = true
+[tikv-client]
+max-batch-wait-time = 2000000
+```
+
+TiKV:
+
+```toml
+log-level = "error"
+[server]
+status-addr = "0.0.0.0:20180"
+grpc-concurrency = 6
+[readpool.storage]
+normal-concurrency = 10
+[rocksdb.defaultcf]
+block-cache-size = "14GB"
+[rocksdb.writecf]
+block-cache-size = "8GB"
+[rocksdb.lockcf]
+block-cache-size = "1GB"
+[raftstore]
+apply-pool-size = 3
+store-pool-size = 3
+```
+
+### TiDB parameter configuration
+
+```
+set global tidb_hashagg_final_concurrency=1;
+set global tidb_hashagg_partial_concurrency=1;
+set global tidb_disable_txn_auto_retry=0;
+```
+
+### Hardware recommendations
+
+#### Machine types
+
+For the test in single AZ (Available Zone), the following machine types are chosen:
+
+| Component     | Instance type       | Count  |
+| :---     | :---------     | :---- |
+| PD       | n1-standard-4  | 3     |
+| TiKV     | c2-standard-16 | 3     |
+| TiDB     | c2-standard-16 | 3     |
+| Sysbench | c2-standard-30 | 1     |
+
+For the test (2019.08) where the result in multiple Azs is compared with that in single AZ, the c2 machine cannot be simultaneously adopted in three AZs within the same GCP Region, so the following machine types are chosen:
```

Comment:
```suggestion
For the test (2019.08) where the result in multiple Azs is compared with that in single AZ, the c2 machine is not simultaneously available in three AZs within the same GCP Region, so the following machine types are chosen:
```

Reply:
@cofyc Is this what you mean?

Reply:
from my perspective, both are correct. the c2 machine is not  simultaneously available in three AZs, so we cannot adopt c2 machine simultaneously in three AZs.

Reply:
however, I prefer the suggested version which clarifies the reason behind.